Library/Homebrew/cmd/info.rb

```diff
@@ -146,6 +146,22 @@ def info_formula(f)
       end
     end
 
+    unless f.requirements.to_a.empty?
+      ohai "Requirements"
+      %w[build required recommended optional].map do |type|
+        reqs = if type == "build"
+          f.requirements.select(&:build?).uniq
+        elsif type == "required"
+          f.requirements.select(&:required?).uniq
+        elsif type == "recommended"
+          f.requirements.select(&:recommended?).uniq
+        elsif type == "optional"
+          f.requirements.select(&:optional?).uniq
+        end
```

Comment:
For that matter, you could replace the entire block with `f.requirements.select(&:"#{type}?")`